### PR TITLE
Add Self-assessment Docs

### PR DESCRIPTION
**Pull request by cailynse**
Pulls in: #70 (I have added some review comments to the parts I cherry-picked to ensure I am understanding correctly)

Given that this is a new subproject and it has been getting some attention we wanted to ensure that there is discoverable information for folks that would like to be involved. 

We have added a basic README to provide some information about the subproject and added a recipe book to help those interested in completing a Self-assessment to get started.

### CODE REVIEW

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [inline sig-security-assessments/Documentation/recipe-book.md:7]: If we are allowing this to be a bit more self serve should we have a dedicated sub-project channel for folks to easily find potential security champions for their assessments? 

We could also have an intake form so project owners could make a formal request to start a Self-assessment?

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [inline sig-security-assessments/Documentation/recipe-book.md:10]: If we are allowing this to be a bit more self serve should we have a dedicated sub-project channel for folks to easily find potential security champions for their assessments? 

We could also have an intake form so project owners could make a formal request to start a Self-assessment?

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [inline sig-security-assessments/Documentation/recipe-book.md:14]: @aladewberry - is this intended for someone in the process of starting a Self-assessment (requester) or is this someone looking for projects that may want a Self-Assessment?

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [inline sig-security-assessments/Documentation/recipe-book.md:31]: I'm flip flopping between the term "security champion" and "security expert" - does anyone have strong opinions one way or another?

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [inline sig-security-assessments/Documentation/recipe-book.md:43]: Are all fields relevant and useful in the TAG-Security Self-Assessment template or would it be useful to create a fork of their template for this project?

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [inline sig-security-assessments/Documentation/recipe-book.md:21]: question: Do we want to do this? Might we just share the assessment slack channel, and maybe an invite to the meeting to the projects slack channel since they may not be on our mailing list?

File: sig-security-assessments/Documentation/recipe-book.md
Comment (PushkarJ) [reply sig-security-assessments/Documentation/recipe-book.md:45]: Fork would be nice especially after we have lessons learnt from vsphere csi-driver assessment too

File: sig-security-assessments/Documentation/recipe-book.md
Comment (PushkarJ) [reply sig-security-assessments/Documentation/recipe-book.md:21]: IMO, Probably okay to skip.

File: sig-security-assessments/README.md
Comment (PushkarJ) [inline sig-security-assessments/README.md:24]:
```suggestion
* Ala Dewberry (**[@aladewberry](https://github.com/aladewberry)**): https://github.com/kubernetes/sig-security/blob/main/sig-security-assessments/OWNERS, VMware
```

For future us, would be easier to know if this needs an update as OWNERS file will be the source of truth on ownership of the subproject always.

**Comment by PushkarJ** [comment]
Really appreciate the welcoming tone of the document. Added some minor comments, but otherwise looks good! Thank you @cailynse for proactively adding comments where reviews maybe needed

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [reply sig-security-assessments/Documentation/recipe-book.md:40]: I'll bring this up in the next meeting.

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [inline sig-security-assessments/Documentation/recipe-book.md:17]: question: At what point do we want Self-assessment project teams to open a request for a slack channel?

File: sig-security-assessments/README.md
Comment (cailynse) [reply sig-security-assessments/README.md]: done!

File: sig-security-assessments/Documentation/recipe-book.md
Comment (aladewberry) [reply sig-security-assessments/Documentation/recipe-book.md:28]: I think Champion is more inviting than Expert. One of the principles here that I want the subproject to espouse is that you don't have to be an expert to do something to positively augment your security posture. Having *some* security knowledge and acting on it is a good thing, in other words. Yes, we absolutely need to partner with experts to make sure that things are properly thought through, thorough, etc. But any non-expert should feel empowered to start a self-assessment and to reach out to experts for guidance, which is what we're here for.

File: sig-security-assessments/Documentation/recipe-book.md
Comment (aladewberry) [reply sig-security-assessments/Documentation/recipe-book.md:17]: Hm - I think once there is a solid group of people who are interested in participating in the self assessment that a slack channel becomes necessary for coordination and publicity/socializing that a self-assessment is underway. /2p

File: sig-security-assessments/Documentation/recipe-book.md
Comment (aladewberry) [reply sig-security-assessments/Documentation/recipe-book.md:10]: There is a dedicated Self Assessment channel already that I think we can totally use to source security champions: https://kubernetes.slack.com/archives/C0441E11REC. Do you think that's sufficient?

The intake form lives here when you add an issue to sig-security: https://github.com/kubernetes/sig-security/issues/new/choose

File: sig-security-assessments/Documentation/recipe-book.md
Comment (cailynse) [reply sig-security-assessments/Documentation/recipe-book.md:43]: I will not do that as part of this PR - but as a follow up. We will fork the TAG-Security one, and then we can iterate on it and make it more suited to our use cases.

Comment (aladewberry) [reply sig-security-assessments/Documentation/recipe-book.md:43]: @cailynse @PushkarJ  just wanted to check in - is there any action I need to take here?

Comment (cailynse) [reply sig-security-assessments/Documentation/recipe-book.md:43]: For the template I'll open a separate PR once this one is merged.

For this PR we need a /lgtm from you and then an approval from a chair I believe!

**Comment by PushkarJ** [comment]
/lgtm

(For posterity and to clarify that I am happy with the changes post review)

File: sig-security-assessments/Documentation/recipe-book.md
Comment (sftim) [inline sig-security-assessments/Documentation/recipe-book.md:7]: (nit!)
```suggestion
1.	Open a Security Assessment Request using the [issue template](https://github.com/kubernetes/sig-security/issues/new/choose).
```

File: sig-security-assessments/Documentation/recipe-book.md
Comment (sftim) [inline sig-security-assessments/Documentation/recipe-book.md:15]: (nit)
```suggestion
1.	To get further participation, advertise in SIG meetings and channels for the project, as well as SIG Security. Aim for a minimum of 4 
```

File: sig-security-assessments/Documentation/recipe-book.md
Comment (sftim) [inline sig-security-assessments/Documentation/recipe-book.md:19]: (nit)
```suggestion
a Slack channel for the project specific assessment - here's an [example](https://github.com/kubernetes/community/pull/7015).
```

File: sig-security-assessments/Documentation/recipe-book.md
Comment (sftim) [inline sig-security-assessments/Documentation/recipe-book.md:20]:
```suggestion
1.	Create a Google doc for meeting notes using the
```

File: sig-security-assessments/Documentation/recipe-book.md
Comment (sftim) [inline sig-security-assessments/Documentation/recipe-book.md:26]: (nit)
```suggestion
- Create a recurring meeting series of about 6 meetings (this is flexible and can be extended if necessary).
```

File: sig-security-assessments/Documentation/recipe-book.md
Comment (sftim) [inline sig-security-assessments/Documentation/recipe-book.md:46]:
```suggestion
- Now is a good time to start writing a [blog post](https://kubernetes.io/docs/contribute/new-content/blogs-case-studies/) to
```

File: sig-security-assessments/README.md
Comment (sftim) [inline sig-security-assessments/README.md:8]:
```suggestion
- [Cluster API](https://github.com/kubernetes/sig-security/blob/1e6f14de2edae1662e2897a933071928515cc27c/sig-security-assessments/cluster-api/self-assessment.md)
```

File: sig-security-assessments/README.md
Comment (sftim) [inline sig-security-assessments/README.md:16]: Is this better?
```suggestion
If you would like to participate in the SIG Security Self-assessment area of work, please join the SIG Security
```

**Review by sftim — COMMENTED** [review]
From a writing perspective, LGTM!

I spotted a few nits you can fix if you'd like to.

**Review by aladewberry — APPROVED** [review]

**Pull request merged**
